daemon/mgr/container.go

```diff
@@ -682,6 +687,79 @@ func (mgr *ContainerManager) Rename(ctx context.Context, oldName, newName string
 	return nil
 }
 
+// Update updates the configurations of a container.
+func (mgr *ContainerManager) Update(ctx context.Context, name string, config *types.UpdateConfig) error {
+	c, err := mgr.container(name)
+	if err != nil {
+		return err
+	}
+
+	c.Lock()
+	defer c.Unlock()
+
+	// update ContainerConfig of a container.
+	if !c.IsStopped() && config.Image != "" || len(config.Env) > 0 {
+		return fmt.Errorf("Only can update the container's image or Env when it is stopped")
+	}
+
+	if config.Image != "" {
+		image, err := mgr.ImageMgr.GetImage(ctx, config.Image)
+		if err != nil {
+			return err
+		}
+		// TODO Image param is duplicate in ContainerMeta
+		c.meta.Config.Image = image.Name
+		c.meta.Image = image.Name
+	}
+
+	if len(config.Env) != 0 {
+		c.meta.Config.Env = config.Env
```

Comment:
Do we just use the config.Env to replace the original one?
For example, if the `c.meta.Config.Env` is {"a": "b"; "c":"d"}, and `config.Env` is {"a": "z"}.

If code goes like your way, the new `c.meta.Config.Env` will be {"a":"z"} rather than {"a": "z", "c": "d"}. Is this reasonable?